### PR TITLE
[ASAP-534] - update quick check details fields

### PR DESCRIPTION
**Pull request by AkosuaA**
This pr updates the quick check details fields on the Manuscript Version content model in the cms to reference a Discussion. In order not to break the app, I've also updated the manuscript data provider to work with the cms changes. The next pr will add the replies functionality and update how we fetch the quick check detail's created by user.

### CODE REVIEW

File: apps/crn-server/src/data-providers/contentful/manuscript.data-provider.ts [hunk 1]
```diff
@@ -285,3 +307,72 @@ const parseComplianceReport = (
...
+    ? getLinkEntity(quickCheckDetails.protocolsDepositedDetails)
+    : null,
+});
+
+const createQuickCheckDiscussions = async (
+  environment: Environment,
+  quickCheckDetails: QuickCheckDetailsObject,
+  userId: string,
+) => {
+  const generatedDiscussions = {} as QuickCheckDetailsObject;
```
Comment (gabiayako) [inline apps/crn-server/src/data-providers/contentful/manuscript.data-provider.ts:347]: I wonder if we could remove `parseQuickCheckDiscussions` doing some modifications like this one:

```suggestion
  const generatedDiscussions = {} as Record<
    QuickCheckDetails,
    Link<'Entry'> | null
  >;
```

and the one in the comment below, what do you think?

File: apps/crn-server/src/data-providers/contentful/manuscript.data-provider.ts [hunk 2]
```diff
@@ -285,3 +307,72 @@ const parseComplianceReport = (
...
+            fields: addLocaleToFields({
+              message: getLinkEntity(messageId),
+            }),
+          });
+
+          await discussionEntry.publish();
+
+          generatedDiscussions[quickCheckDetail as QuickCheckDetails] =
+            discussionEntry.sys.id;
+        }
```
Comment (gabiayako) [inline apps/crn-server/src/data-providers/contentful/manuscript.data-provider.ts:373]:
```suggestion
          generatedDiscussions[quickCheckDetail as QuickCheckDetails] =
            getLinkEntity(discussionEntry.sys.id);
        } else {
          generatedDiscussions[quickCheckDetail as QuickCheckDetails] = null;
        }
```

**Review by gabiayako — APPROVED** [review]
nice

File: apps/crn-server/src/data-providers/contentful/manuscript.data-provider.ts (hunk 1 above)
Comment (AkosuaA) [reply apps/crn-server/src/data-providers/contentful/manuscript.data-provider.ts:347]: Yea, it simplifies the steps. Thanks! 👌🏾

**Pull request merged**